### ISSUE TITLE
Is no-irregular-whitespace fixable?

### ISSUE THREAD

**Issue by balazsorban44** [eslint/eslint#12846]
I recently came across an issue in VS Code.
I have `no-irregular-whitespace` enabled through `"extends": "eslint:recommended"`.

When writing functions, I often come across `AltGr + Space`, which results in an irregular whitespace. It is nice to get a warning about it, but I was curious if it is an issue that can be fixed automatically by this rule - meaning when found, replace the irregular whitespace character with a regular one - or there is a good reason it is not implemented?
![image](https://user-images.githubusercontent.com/18369201/73433689-7d2b1000-4345-11ea-9b89-9bdb8014fae3.png)

**What rule do you want to change?**
[no-irregular-whitespace](https://eslint.org/docs/rules/no-irregular-whitespace)
**Does this change cause the rule to produce more or fewer warnings?**
I am not sure, I guess after fixing the problem, it will cause ESLint to produce fewer warnings.
**How will the change be implemented? (New option, new default behavior, etc.)?**
If I [understand it correctly](https://eslint.org/docs/developer-guide/working-with-rules#rule-basics), one has to add a  `fixable` property to the `meta` object in the Rule Definition, and probably implement some code.
**Please provide some example code that this change will affect:**

```js
function foo()/*irregular whitespace here*/{}
// after fix ->
function foo() {}
```

**What does the rule currently do for this code?**
It reports the problem, but do not fix it automatically

**What will the rule do after it's changed?**
If configured, it will report AND fix the problem, meaning replacing any irregular whitespace character with a regular space character.
**Are you willing to submit a pull request to implement this change?**
I am afraid I don't have a knowledge by myself, but If I get some support, and it is something that CAN be fixed, I am willing/happy to try.

**Comment by kaicataldo** [comment]
The only situation I can think of where this might be a problem is that it could result in generating new errors that [`no-mixed-spaces-and-tabs`](https://eslint.org/docs/rules/no-mixed-spaces-and-tabs) would report. That being said, it seems like it would be useful to me!

I wonder if it would make sense to add a configuration option for the autofixer to tell the rule to replace with spaces or tabs. I don't _think_ we have a precedence for configuration options for autofixers, but I think it's worth considering for the reason above.

Alternatively, `no-mixed-spaces-and-tabs` could add an autofixer with the above configuration instead, and then a multi-pass autofix run would fix all errors.

**Comment by mdjermanovic** [comment]
I'm not too familiar with this subject, but could the replacement sometimes (or maybe always?) depend on the particular irregular character? For example, some of those seem to be linebreaks.

Also, would it make sense to auto-fix for the start just NBSP as it seems to be the most frequent problem, and replace it always with just a "regular" space.
